### PR TITLE
Migrate to GitHub Actions, use @lavamoat/allow-scripts

### PR DESCRIPTION
**Pull request by rekmarks**
Adds `@lavamoat/allow-scripts` and migrates from CircleCI to GitHub Actions. The module-template `build-lint-test` workflow has been adapted for this repository.

Actions have been enabled in settings.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +16,7 @@
   ],
   "devDependencies": {
     "@eslint/eslintrc": "^0.4.0",
+    "@lavamoat/allow-scripts": "^1.0.6",
```
Comment (Gudahtt) [inline package.json:19]: This isn't required - it's a transitive dependency that isn't used directly.

```suggestion
```

Comment (rekmarks) [reply package.json:19]: Indeed!

Comment (Gudahtt) [reply package.json]: Wait what.... I could have sworn I made this comment on the line importing `preinstall-always-fail`. Now it's saying I commented on the `allow-scripts` dependency. If this is in fact what I did, I apologize. We do need `allow-scripts` as a dependency.

Comment (rekmarks) [reply package.json]: lol, I was thinking the same thing. I'll fix it

Comment (rekmarks) [reply package.json:19]: I _didn't_ have `preinstall-always-fail` in the lockfile at all. If we're both experiencing the same hallucination, is it really a hallucination?

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by Gudahtt** [comment]
I've just disabled the CircleCI status checks as well, so no more red X's after this is merged

**Pull request merged**